StudyHub/StudyHub.BLL/Services/Auth/AuthService.cs

```diff
@@ -1,77 +1,67 @@
 using StudyHub.Common.Exceptions;
-using StudyHub.DAL.Repositories.Interfaces;
 using StudyHub.Entities;
 using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Identity;
+using StudyHub.BLL.Services.Interfaces.Auth;
+using AutoMapper;
 
-namespace StudyHub.BLL.Services;
+namespace StudyHub.BLL.Services.Auth;
 
 // ToDo: Implement Refresh Token
 public class AuthService : IAuthService
 {
-    private readonly UserManager<User> _userManager;
-    private readonly ITokenService _tokenService;
-    private readonly IRepository<InvitedUser> _invitedUserRepository;
-    private readonly IMapper _mapper;
+    protected readonly UserManager<User> _userManager;
+    protected readonly ITokenService _tokenService;
+    protected readonly IMapper _mapper;
 
     public AuthService(
         UserManager<User> userManager,
         ITokenService tokenService,
-        IRepository<InvitedUser> invitedUserRepository,
         IMapper mapper)
     {
-        _mapper = mapper;
-        _invitedUserRepository = invitedUserRepository;
         _userManager = userManager;
         _tokenService = tokenService;
+        _mapper = mapper;
     }
 
-    public async Task<AuthSuccessDTO> LoginAsync(LoginUserDTO user)
+    public async Task<AuthSuccessDTO> LoginAsync(LoginUserDTO dto)
     {
-        var existingUser = await _userManager.FindByEmailAsync(user.Email)
-            ?? throw new NotFoundException($"Unable to find user by specified email. Email: {user.Email}");
+        var user = await _userManager.FindByEmailAsync(dto.Email);
+
+        if (user == null)
+            throw new NotFoundException($"Unable to find user by specified email. Email: {dto.Email}");
 
-        var isPasswordValid = await _userManager.CheckPasswordAsync(existingUser, user.Password);
+        var isPasswordValid = await _userManager.CheckPasswordAsync(user, dto.Password);
 
         if (!isPasswordValid)
-            throw new InvalidCredentialsException($"User input incorrect password. Password: {user.Password}");
+            throw new InvalidCredentialsException($"User input incorrect password. Password: {dto.Password}");
 
-        return await GetAuthTokens(existingUser);
+        return await GetAuthTokensAsync(user);
     }
 
     public async Task<AuthSuccessDTO> RegisterAsync(RegisterUserDTO dto)
     {
-        var invitedUser = _invitedUserRepository.FirstOrDefault(e => e.Email == dto.Email)
-            ?? throw new NotFoundException($"User with this email wasn't invited: {dto.Email}");
+        var user = await _userManager.FindByEmailAsync(dto.Email);
 
-        if (invitedUser.Token != dto.Token)
-            throw new IncorrectParametersException("Passed token isn't valid.");
+        if (user != null)
+            throw new NotFoundException($"User with specified email already exists. Email: {dto.Email}");
 
-        var user = _mapper.Map<User>(dto);
+        user = _mapper.Map<User>(dto);
```

Comment:
it seems weird for me that we used one ,,user,, for validation and, subsequently, registaring them

Reply:
Why should we create another one?